Euclidean Clustering can optionally publish filtered point cloud and/or filtered ground points

amc-nu opened the pull request:
## Modified euclidean clustering to:
1. Publish new topic "/points_ground" of the type sensor_msgs::PointCloud2, outputs the planar points in the ground
2. Publish new topic "/points_filtered" of the type sensor_msgs::PointCloud2, removes the planar points from points_raw
   Both of the added features feed from the customizable 'points_node' argument.
   **Please check the launch file for details.**

The idea is to generate different pointcloud messages to be projected using the new points2image.
#### Example:
- Generate the PC messages

```
% roslaunch lidar_tracker euclidean_clustering.launch
```

This will publish 3 topics, /euclidean_clustering, /points_filtered, /points_ground
- Launch calibration_publisher
- Launch points2image to show the projected result from the desired PC message
  For instance:

```
% rosrun points2image points2image _points_node:=/points_filtered
```

or

```
% rosrun points2image points2image _points_node:=/points_ground
```

etc...
Finally :

```
% rosrun viewers points_image_viewer
```

In summary:
Any PointCloud2 Message --->  Points2Image --->  Viewer

Could @h-ohta or @manato, confirm please with a real rosbag?
**Remember to (1)relay velodyne_points to points_raw, (2)Publish camera_info and the projection matrix, (3) run points2image with /points_filtered or /points_ground**

@syohex Could you please check the correct compilation of this branch? 

Thanks for your kind support

![Image of Euclidean](http://ertl.jp/~amonrroy/images/euc1.png)
![Image of Euclidean2](http://ertl.jp/~amonrroy/images/euc2.png)

amc-nu committed:
-Modified euclidean clustering to:
1. publish new topic "/points_ground" of the type sensor_msgs::PointCloud2, outputs the planar points in the ground
2. publish new topic "/points_filtered" of the type sensor_msgs::PointCloud2, removes the planar points from points_raw
Both of the added features feed from the customizable 'points_node' argument.
Please check the launch file for details.

The idea is to generate different pointcloud messages to be projected using the new points2image.
Example:
1. Generate the PC messages
% roslaunch lidar_tracker euclidean_clustering.launch
This will publish 3 topics, /euclidean_clustering, /points_filtered, /points_ground

2. Launch calibration_publisher

3. Launch points2image to show the projected result from the desired PC message
For instance:
% rosrun points2image points2image _points_node:=/points_filtered
or
% rosrun points2image points2image _points_node:=/points_ground
etc...

4. finally :
% rosrun viewers points_image_viewer

Any PointCloud2 Message --->  Points2Image --->  Viewer

syohex commented on ros/src/computing/perception/detection/packages/lidar_tracker/nodes/euclidean_cluster/euclidean_cluster.cpp:
```diff
@@ -211,6 +229,18 @@ int main (int argc, char** argv)
+	if (private_nh.getParam("publish_ground", publish_ground))
+	{
+		ROS_INFO("Publishing /points_ground point cloud...");
+		pub_filtered = h.advertise<sensor_msgs::PointCloud2>("/points_filtered",1);
```
`ROS_INFO` says publishing **/points_Ground**, but this argument is **/points_filtered**.
Why ?

And you use absolute topic path, is this right ?

syohex commented on ros/src/computing/perception/detection/packages/lidar_tracker/nodes/euclidean_cluster/euclidean_cluster.cpp:
```diff
@@ -211,6 +229,18 @@ int main (int argc, char** argv)
+	if (private_nh.getParam("publish_filtered", publish_filtered))
+	{
+		ROS_INFO("Publishing /points_filtered point cloud...");
+		pub_ground = h.advertise<sensor_msgs::PointCloud2>("/points_ground",1);
```
Ditto.

Commit:
Make pointCloud topic selectable subscribed in points_image_viewer

amc-nu committed:
as pointed by Yosh

regarding the topics name, I'm just following the convention used in the file. (ie. "/points_cluster")

amc-nu committed:
Update euclidean_clustering.launch
